h/static/scripts/streamer.js

```diff
@@ -96,64 +86,78 @@ function connect($rootScope, annotationMapper, groups, session, settings) {
-    var websocketHost = new URL(url).hostname;
-    if (['localhost', '127.0.0.1'].indexOf(websocketHost) !== -1) {
-      console.warn('Check that your H service is configured to allow ' +
-                   'WebSocket connections from ' + window.location.origin);
+    // If we have no URL configured, don't do anything.
+    var url = settings.websocketUrl;
+    if (!url) {
+      return;
     }
-  });
-
-  socket.on('message', function (event) {
-    // Wrap message dispatches in $rootScope.$apply() so that
-    // scope watches on app state affected by the received message
-    // are updated
-    //
-    // Note: The use of $apply() here will no longer be needed once session
-    // state is moved to the Redux store in `annotationUI`.
-    $rootScope.$apply(function () {
-      var message = JSON.parse(event.data);
-      if (!message) {
-        return;
+
+    socket = new Socket(url);
+    socket.on('open', function () {
+      sendClientConfig();
+    });
+
+    socket.on('error', function (event) {
+      console.warn('Error connecting to H push notification service:', event);
+
+      // In development, warn if the connection failure might be due to
+      // the app's origin not having been whitelisted in the H service's config.
+      //
+      // Unfortunately the error event does not provide a way to get at the
+      // HTTP status code for HTTP -> WS upgrade requests.
+      var websocketHost = new URL(url).hostname;
+      if (['localhost', '127.0.0.1'].indexOf(websocketHost) !== -1) {
+        console.warn('Check that your H service is configured to allow ' +
+                     'WebSocket connections from ' + window.location.origin);
       }
+    });
+
+    socket.on('message', function (event) {
+      // wrap message dispatches in $rootScope.$apply() so that
+      // scope watches on app state affected by the received message
+      // are updated
+      $rootScope.$apply(function () {
+        var message = JSON.parse(event.data);
+        if (!message) {
+          return;
+        }
+
+        if (message.type === 'annotation-notification') {
+          handleAnnotationNotification(message);
+        } else if (message.type === 'session-change') {
+          handleSessionChangeNotification(message);
+        } else {
+          console.warn('received unsupported notification', message.type);
+        }
+      });
+    });
 
-      if (message.type === 'annotation-notification') {
-        handleAnnotationNotification(message);
-      } else if (message.type === 'session-change') {
-        handleSessionChangeNotification(message);
-      } else {
-        console.warn('received unsupported notification', message.type);
+    var self = this;
+    var sidebarOpen = false;
+    $rootScope.$on('sidebarOpened', function () {
+      sidebarOpen = true;
+      if (!socket) {
+        self.connect();
+      }
+    });
+    $rootScope.$on(events.USER_CHANGED, function () {
+      if (sidebarOpen) {
+       self.connect();
```

Comment:
I forgot why this was needed initially. So the reason for it is to disconnect and then re-connect the WebSocket after the user's authentication status changes.